src/librustc_mir/interpret/place.rs

```diff
@@ -1082,17 +1082,21 @@ where
                 }
                 if variant_index != dataful_variant {
                     let variants_start = niche_variants.start().as_u32();
-                    let variant_index_relative = variant_index.as_u32()
-                        .checked_sub(variants_start)
-                        .expect("overflow computing relative variant idx");
```

Comment:
In codegen this isn't even checked, it's a plain `-`, because this is impossible for *indices*.